src/chatgpt.rs

```diff
@@ -33,6 +33,20 @@ pub struct ChatResponse {
     choices: Vec<Choice>,
 }
 
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ErrorResponse{
+    pub error: ErrorDetail,
+}
+
+#[derive(Debug, Serialize, Deserialize)]
+pub struct ErrorDetail{
+    message: String,
+    #[serde(default)]
+    type_:String,
+    param:Option<String>,
+    code:Option<String>,
```

Comment:
🌱 Serde is acutally okay with you not mapping out the entire json object. So here we actually only need the message property since it is the only one we are making use of in the code, this struct can actually just be:

```rust
#[derive(Debug, Serialize, Deserialize)]
pub struct ErrorDetail {
    message: String,
}
```

Reply:
Great to know, was trying to figure out how to make it more brief but wasn't sure what was necessary. Thanks for taking the time to review.